### PR TITLE
feat: add jsonista support

### PR DESCRIPTION
**Pull request by rschmukler**
Adds support for Metosin's jsonista. Mostly just a copy paste of the other implementations. Thanks for all your work on the library (and the ecosystem in general!)

### CODE REVIEW

File: src/babashka/json.clj [hunk 1]
```diff
@@ -16,14 +16,17 @@
...
+                  'babashka.json.internal.jsonista/fns)))
         (or
          ;; this should always work in babashka
          (try-require 'babashka.json.internal.cheshire/fns)
          ;; this needs to be added to the classpath explicitly
          (try-require 'babashka.json.internal.charred/fns)
          ;; this one should always work as this project has a dependency on it.
-         (try-require 'babashka.json.internal.data-json/fns)))]
+         (try-require 'babashka.json.internal.data-json/fns)
+         (try-require 'babashka.json.internal.jsonista/fns)))]
```
Comment (borkdude) [inline src/babashka/json.clj]: The jsonista try-require should go before data.json, as data.json always succeeds

File: src/babashka/json/internal/jsonista.clj [hunk 2]
```diff
@@ -0,0 +1,21 @@
+(ns babashka.json.internal.jsonista
+  (:require [jsonista.core :as j]))
+
+(defn- -key-fn->object-mapper
+  [key-fn]
+  (if-not key-fn
+    j/default-object-mapper
+    (j/object-mapper
+      {:encode-key-fn key-fn
```
Comment (borkdude) [inline src/babashka/json/internal/jsonista.clj]: If one would pass `keyword` as the keyfn, wouldn't it be weird to encode a keyword using `keyword`?

Comment (rschmukler) [reply src/babashka/json/internal/jsonista.clj]: It's not actually used, but I can drop it (ie. there's no actual way to pass a `key-fn` into `write-str`.

Comment (borkdude) [reply src/babashka/json/internal/jsonista.clj]: Yes, I think it'd be better to drop it for now, until we explicitly add support for it.

**Comment by rschmukler** [comment]
@borkdude forced push w/ requested changes, thanks again!

**Comment by rschmukler** [comment]
Also, test actions are running on my fork: https://github.com/rschmukler/json/actions/runs/4356270102

**Comment by borkdude** [comment]
Thanks!

**Pull request merged**